tests/components/matter/test_water_heater.py

```diff
@@ -0,0 +1,244 @@
+        value=5200,
+    )
+    matter_client.write_attribute.reset_mock()
+
+    @pytest.mark.parametrize("node_fixture", ["silabs_water_heater"])
+    @pytest.mark.parametrize(
+        ("operation_mode", "matter_attribute_value"),
+        [(STATE_OFF, 0), (STATE_ECO, 4), (STATE_HIGH_DEMAND, 4)],
+    )
+    async def test_water_heater_set_operation_mode(
+        hass: HomeAssistant,
+        matter_client: MagicMock,
+        matter_node: MatterNode,
+        operation_mode: str,
+        matter_attribute_value: int,
+    ) -> None:
+        """Test water_heater set operation mode service."""
+        state = hass.states.get("water_heater.water_heater")
+        assert state
+
+        # test change mode to each operation_mode
+        await hass.services.async_call(
+            "water_heater",
+            "set_operation_mode",
+            {
+                "entity_id": "water_heater.water_heater",
+                "operation_mode": operation_mode,
+            },
+            blocking=True,
+        )
+
+        state = hass.states.get("water_heater.water_heater")
+        assert state.state == operation_mode
+        assert matter_client.write_attribute.call_count == 1
+        assert matter_client.write_attribute.call_args == call(
+            node_id=matter_node.node_id,
+            attribute_path=create_attribute_path_from_attribute(
+                endpoint_id=2,
+                attribute=clusters.Thermostat.Attributes.SystemMode,
+            ),
+            value=matter_attribute_value,
+        )
+
+    @pytest.mark.parametrize("node_fixture", ["silabs_water_heater"])
+    async def test_water_heater_boostmode(
+        hass: HomeAssistant,
+        matter_client: MagicMock,
+        matter_node: MatterNode,
+    ) -> None:
+        """Test water_heater set operation mode service."""
+        # Boost 1h (3600s)
+        boost_info: type[
+            clusters.WaterHeaterManagement.Structs.WaterHeaterBoostInfoStruct
+        ] = clusters.WaterHeaterManagement.Structs.WaterHeaterBoostInfoStruct(
+            duration=3600
+        )
+        state = hass.states.get("water_heater.water_heater")
+        assert state
+
+        # enable water_heater boostmode
+        await hass.services.async_call(
+            "water_heater",
+            "set_operation_mode",
+            {
+                "entity_id": "water_heater.water_heater",
+                "operation_mode": STATE_HIGH_DEMAND,
+            },
+            blocking=True,
+        )
+
+        state = hass.states.get("water_heater.water_heater")
+        assert state.state == STATE_HIGH_DEMAND
+        assert matter_client.write_attribute.call_count == 1
+        assert matter_client.write_attribute.call_args == call(
+            node_id=matter_node.node_id,
+            attribute_path=create_attribute_path_from_attribute(
+                endpoint_id=2,
+                attribute=clusters.Thermostat.Attributes.SystemMode,
+            ),
+            value=0,
```

Comment:
How can this be 0 here when it's 4 in the `test_water_heater_set_operation_mode` test?